### ISSUE TITLE
zfs-2.1.15 don't compile with RHEL 9.5 last kernel-5.14.0-503.11.1.el9_5.x86_64

### ISSUE THREAD

**Issue by rgessner** [openzfs/zfs#16746]
### System information

Distribution Name	| Red Hat Enterprise Linux (not a rhel clone)
Distribution Version	| 9.5
Kernel Version	| kernel-5.14.0-503.11.1.el9_5.x86_64
Architecture	| x86_64
OpenZFS Version	| 2.1.15

### Describe the problem you're observing

Working zfs on RHEL 9.4 system with kernel-5.14.0-427.42.1.el9_4.x86_64.
couple of error messages during kernel update (see below).
After booting no zfs module present

### Describe how to reproduce the problem
Update from RHEL 9.4 kernel 5.14.0-427 to RHEL 9.5 kernel-5.14.0.-503

### Include any warning/errors/backtraces from the system logs
```
checking whether blkdev_get_by_path() exists and takes 3 args... no
checking whether blkdev_get_by_path() exists and takes 4 args... no
checking whether bdev_open_by_path() exists... configure: error: 
	*** None of the expected "blkdev_get_by_path()" interfaces were detected.
	*** This may be because your kernel version is newer than what is
	*** supported, or you are using a patched custom kernel with
	*** incompatible modifications.
	***
	*** ZFS Version: zfs-2.1.15-3
	*** Compatible Kernels: 3.10 - 6.7
	

Cleaning build area...(bad exit status: 2)
Failed command:
make -C /lib/modules/5.14.0-503.11.1.el9_5.x86_64/build M=/var/lib/dkms/zfs/2.1.15/build cleanBuilding module(s)...(bad exit status: 2)
Failed command:
make -j4 KERNELRELEASE=5.14.0-503.11.1.el9_5.x86_64
Error! Bad return status for module build on kernel: 5.14.0-503.11.1.el9_5.x86_64 (x86_64)
Consult /var/lib/dkms/zfs/2.1.15/build/make.log for more information.
Autoinstall on 5.14.0-503.11.1.el9_5.x86_64 failed for module(s) zfs(10).

Error! One or more modules failed to install during autoinstall.
Refer to previous errors for more information.

```

**Comment by AllKind** [comment]
I don't think ZFS 2.1.x is not maintained any more.
Could you try with 2.2.6?

**Comment by AllKind** [comment]
You could just run `configure` to see if the check passes. Which could also help others.

**Comment by wanghz10** [comment]
Version 2.2.6 seems to be working. You can enable zfs-testing to install.

dnf config-manager --enable zfs-testing
dnf upgrade
dkms autoinstall

**Comment by llagendijk** [comment]
The testing repo for RHEL9 has 2.2.6 and works for me

**Comment by ItalyPaleAle** [comment]
I had the same issue when updating RHEL 9 to the latest version.

- There doesn't seem to be any package in zfs or zfs-kmod available
- Using DKMS from zfs-testing (2.2.6) worked

**Comment by Germano0** [comment]
when 2.2.x will be moved from "testing" to "stable" repository for RHEL?

**Comment by hawkdavis** [comment]
yeah guys my plex server was down for an entire 10 minutes... unacceptable.

**Comment by tcouey** [comment]
> yeah guys my plex server was down for an entire 10 minutes... unacceptable.

lol. That's how I discovered the problem this morning too... but mine had been down for over 8 hours. I thought I was the only one running Plex on RHEL!

**Comment by tonyhutter** [comment]
Thanks all for reporting this.  Looks like the new kernel has hit AlmaLinux 9.  I've put together a new release PR with the commits needed to get it building again: https://github.com/openzfs/zfs/pull/16781

**Comment by Germano0** [comment]
> Looks like the new kernel has hit AlmaLinux 9

@tonyhutter you can use RHEL for free under certain circumstances (see terms at https://developers.redhat.com/articles/faqs-no-cost-red-hat-enterprise-linux )

**Comment by mtelka** [comment]
I assume this is a duplicate of #16196.

**Comment by cyqsimon** [comment]
And PSA: don't forget to `dnf config-manager --disable zfs-testing` after you're done. Unless you actually want to be on the rolling channel, which you most likely don't.

**Comment by principis** [comment]
> Thanks all for reporting this. Looks like the new kernel has hit AlmaLinux 9. I've put together a new release PR with the commits needed to get it building again: #16781

This does not seem to work. Is there anything else that's needed?

**Comment by makhomed** [comment]
> Thanks all for reporting this. Looks like the new kernel has hit AlmaLinux 9. I've put together a new release PR with the commits needed to get it building again: #16781

AlmaLinux OS is not 1:1 RHEL clone, AlmaLinux OS Foundation board Jul 13, 2023 has decided to drop the aim to be 1:1 with RHEL:

> Now that we will no longer be holding ourselves to being a 1:1 Red Hat downstream rebuild, we are taking some time to consider the possibilities around what that means. We will continue to provide updates around that process and will include the members of the AlmaLinux OS Foundation in that conversation and decision-making process as well. -- https://almalinux.org/blog/future-of-almalinux/

It is more safe to use original RHEL - it is no-cost for development use - https://developers.redhat.com/articles/faqs-no-cost-red-hat-enterprise-linux

Or use [Rocky Linux](https://rockylinux.org/), because [Rocky Linux](https://rockylinux.org/) is designed to be 100% bug-for-bug compatible with RHEL.

**Comment by deajan** [comment]
As far as I understand ZoL supports current and last release, so since we're not far away from OpenZFS 2.3, we safely can enable 2.2.6 on current EL9 since 2.2.x branch will become default for EL9 in a not so far distant feature.
@tonyhutter Can you confirm perhaps ?

**Comment by TopherIsSwell** [comment]
> ```shell
> dnf update --enablerepo=zfs-testing zfs
> ```

This worked for me on both Rocky 9 and Alma 9. But what was interesting was that building 2.2.6 and 2.3.0-RC3 from source (actually just running the configure script) on Alma 9 failed with the same error, but installing from the zfs-testing repo worked fine. Is there some kind of secret sauce in the RPM that's not accessible by doing the following in the source?
```bash
./autogen.sh
./configure
```



> It is more safe to use original RHEL - it is no-cost for development use - https://developers.redhat.com/articles/faqs-no-cost-red-hat-enterprise-linux
Using the RHEL development license is as easy as 1, 2, 3 hours of searching their website for the correct process.

**Comment by Germano0** [comment]
we can close this ticket

**Comment by dennixxNL** [comment]
The on 6 December released 2.1.16 seems to work well on RHEL9

**Comment by deajan** [comment]
Confirmed.
